### PR TITLE
AJ-984: fix console errors on Data tab

### PR DESCRIPTION
**Pull request by davidangb**
## Summary of changes:
Resolves console errors related to use of `Utils.cond` on the Data tab.

Attention Reviewer: I've also reformatted some code for readability. You may want to ignore whitespace when viewing the diff.

### Testing strategy
Tested locally.

This branch:
Screenshot 2023-04-30 at 3 38 57 PM

Dev environment:
Screenshot 2023-04-30 at 3 38 38 PM

### CODE REVIEW

File: src/pages/workspaces/workspace/Data.js [hunk 1]
```diff
@@ -1064,89 +1067,94 @@ const WorkspaceData = _.flow(
...
-        )], () => div({ style: { textAlign: 'center', lineHeight: '1.4rem', marginTop: '1rem', marginLeft: '5rem', marginRight: '5rem' } }, ['Currently, only a workspace creator can use data tables. If you were invited to this workspace, you can use the clone feature to create your own workspace to use data tables.']))
+        Utils.cond(
+          [createdBy === getUser()?.email || isGoogleWorkspace, () => Utils.switchCase(selectedData?.type,
+            [undefined, () => Utils.cond(
+              [isAzureWorkspace && !wdsReady, () => div({ style: { textAlign: 'center', lineHeight: '1.4rem', marginTop: '1rem', marginLeft: '5rem', marginRight: '5rem' } },
+                [icon('loadingSpinner'),
+                  ' Preparing your data tables, this may take a few minutes. ',
+                  Utils.cond(
+                    [!uploadingWDSFile, () => div({}, ['You can ', h(Link, { style: { marginTop: '0.5rem' }, onClick: () => setTroubleshootingWds(true) }, ['check the status']), ' of your data table service.'])],
+                    () => () => '')])],
```
Comment (davidangb) [inline src/pages/workspaces/workspace/Data.js]: the only non-whitespace change from lines 1067-1078 is a change to the final catch-all case of the `Utils.cond`.
I changed `''` to `() => ''`.

File: src/pages/workspaces/workspace/Data.js [hunk 2]
```diff
@@ -1064,89 +1067,94 @@ const WorkspaceData = _.flow(
...
+            [workspaceDataTypes.wds, () => wdsDataTableProvider && wdsReady && !_.isEmpty(wdsTypes.state) && h(WDSContent, {
+              key: refreshKey,
+              workspaceUUID: workspaceId,
+              workspace,
+              dataProvider: wdsDataTableProvider,
+              recordType: selectedData.entityType,
+              wdsSchema: wdsTypes.state
+            })]
+          )],
+          () => div({ style: { textAlign: 'center', lineHeight: '1.4rem', marginTop: '1rem', marginLeft: '5rem', marginRight: '5rem' } }, ['Currently, only a workspace creator can use data tables. If you were invited to this workspace, you can use the clone feature to create your own workspace to use data tables.']))
```
Comment (davidangb) [inline src/pages/workspaces/workspace/Data.js:1157]: this is a whitespace-only change.

File: src/pages/workspaces/workspace/Data.js [hunk 3]
```diff
@@ -748,7 +748,10 @@ const WorkspaceData = _.flow(
             ])
           }, [h(ButtonOutline, {
             disabled: !canEditWorkspace || uploadingWDSFile,
-            tooltip: Utils.cond([uploadingWDSFile, 'Upload in progress'], canEditWorkspace ? 'Add data to this workspace' : editWorkspaceErrorMessage),
+            tooltip: Utils.cond(
+              [uploadingWDSFile, () => 'Upload in progress'],
+              () => (canEditWorkspace ? 'Add data to this workspace' : editWorkspaceErrorMessage)
+            ),
```
Comment (davidangb) [inline src/pages/workspaces/workspace/Data.js:754]: both conditions in this `Utils.cond` have actual syntax changes, as well as adding linebreaks.

**Review by calypsomatic — APPROVED** [review]

File: src/pages/workspaces/workspace/Data.js [hunk 4]
```diff
@@ -1064,89 +1067,94 @@ const WorkspaceData = _.flow(
...
-        )], () => div({ style: { textAlign: 'center', lineHeight: '1.4rem', marginTop: '1rem', marginLeft: '5rem', marginRight: '5rem' } }, ['Currently, only a workspace creator can use data tables. If you were invited to this workspace, you can use the clone feature to create your own workspace to use data tables.']))
+        Utils.cond(
+          [createdBy === getUser()?.email || isGoogleWorkspace, () => Utils.switchCase(selectedData?.type,
+            [undefined, () => Utils.cond(
+              [isAzureWorkspace && !wdsReady, () => div({ style: { textAlign: 'center', lineHeight: '1.4rem', marginTop: '1rem', marginLeft: '5rem', marginRight: '5rem' } },
+                [icon('loadingSpinner'),
+                  ' Preparing your data tables, this may take a few minutes. ',
+                  Utils.cond(
+                    [!uploadingWDSFile, () => div({}, ['You can ', h(Link, { style: { marginTop: '0.5rem' }, onClick: () => setTroubleshootingWds(true) }, ['check the status']), ' of your data table service.'])],
+                    () => '')])],
```
Comment (davidangb) [inline src/pages/workspaces/workspace/Data.js:1078]: the only non-whitespace change from lines 1067-1078 is a change to the final catch-all case of the Utils.cond.
I changed `''` to `() => ''`.

**Review by yuliadub — APPROVED** [review]

**Pull request merged**